<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.361">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Methylation_DNA - Metilación</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./index.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
</head>
<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./Met_ADN.html">Metilación</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Methylation_DNA</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Bienvenido</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Met_ADN.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Metilación</span></a>
  </div>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<h1 class="title">Metilación</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header><p>Se utiliza este codigo para analizar y visulizar los datos de ADN metilado</p>
<div class="cell">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#Loading</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://www.stats.ox.ac.uk/pub/MASS4/">MASS</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://readxl.tidyverse.org">readxl</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ardata-fr.github.io/flextable-book/">flextable</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://www.r-project.org">moments</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span> <span class="op">(</span><span class="va"><a href="https://ggobi.github.io/ggally/">GGally</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com">patchwork</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span>            <span class="co"># Cargar el paquete gridExtra</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/BioinformaticsFMRP/TCGAbiolinks">TCGAbiolinks</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">IlluminaHumanMethylation450kanno.ilmn12.hg19</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">IlluminaHumanMethylation450kmanifest</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hansenlab/minfi">minfi</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/limma/">limma</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">missMethyl</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">DMRcate</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ivanek/Gviz">Gviz</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/edgeR/">edgeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/zwdzwd/sesame">sesame</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/UCSCXenaTools/">UCSCXenaTools</a></span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">data.met</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"data.met.RDS"</span><span class="op">)</span></span>
<span><span class="co"># met matrix # SummarizedExperiment::assay(data.met): Esta parte del código accede </span></span>
<span><span class="co"># a la matriz de datos almacenada en el objeto data.met utilizando la función assay() </span></span>
<span><span class="co"># a los datos de metilación del ADN en un objeto SummarizedExperiment a un objeto de tipo data.frame</span></span>
<span><span class="co"># la salida es una dataframe de el codigo de cad muestra y identificador del sitio CpG específico que está en relación con la metilación del ADN</span></span>
<span><span class="va">met</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">data.met</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># clinical data  se utiliza para extraer y convertir los metadatos clínicos asociados</span></span>
<span><span class="co"># describcion de cada pasient </span></span>
<span><span class="va">clinical1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">data.met</span><span class="op">@</span><span class="va">colData</span><span class="op">)</span> <span class="co">#Esta parte del código accede a la información de los metadatos clínicos almacenados en el objeto data.met utilizando el atributo colData</span></span>
<span><span class="co"># get the 450k annotation data e utiliza para obtener la información de anotación para el array de metilación del ADN Illumina HumanMethylation450k en la versión del genoma hg19.</span></span>
<span><span class="va">ann450k</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/minfi/man/getAnnotation.html">getAnnotation</a></span><span class="op">(</span><span class="va">IlluminaHumanMethylation450kanno.ilmn12.hg19</span><span class="op">)</span></span>
<span><span class="co">## remove probes with NA se utiliza para calcular el número de valores faltantes (NA) en cada fila del objeto</span></span>
<span><span class="va">probe.na</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">met</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html">table</a></span><span class="op">(</span><span class="va">probe.na</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 FALSE   TRUE 
156211 329366 </code></pre>
</div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#FALSE   TRUE </span></span>
<span><span class="co">#103553 382024 </span></span>
<span><span class="co"># chose those has not NA values in rows</span></span>
<span><span class="co"># Para filtrar y seleccionar las filas del objeto met que no tienen valores faltantes, con el objetivo de trabajar solo con los datos completos para análisis posteriores.</span></span>
<span><span class="va">probe</span> <span class="op">&lt;-</span> <span class="va">probe.na</span><span class="op">[</span><span class="va">probe.na</span> <span class="op">==</span> <span class="fl">0</span><span class="op">]</span></span>
<span><span class="co"># para seleccionarlas filas que no tienen valores faltantes.</span></span>
<span><span class="co">## la salida es una dataframe de met aniterior pero sin valores faltante</span></span>
<span><span class="va">met</span> <span class="op">&lt;-</span> <span class="va">met</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">met</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">probe</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="co">## remove probes that match to chromosome  X and Y </span></span>
<span><span class="co"># para eliminar las filas que corresponden a los cromosomas X e Y y mantiene solo las filas que contienen información de otros cromosomas.</span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="op">!</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">met</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html">%in%</a></span> <span class="va">ann450k</span><span class="op">$</span><span class="va">Name</span><span class="op">[</span><span class="va">ann450k</span><span class="op">$</span><span class="va">chr</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"chrX"</span>,<span class="st">"chrY"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html">table</a></span><span class="op">(</span><span class="va">keep</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>keep
 FALSE   TRUE 
  6284 323082 </code></pre>
</div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">met</span> <span class="op">&lt;-</span> <span class="va">met</span><span class="op">[</span><span class="va">keep</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">keep</span><span class="op">)</span> <span class="co"># remove no further needed probes.</span></span>
<span></span>
<span><span class="co">## remove SNPs overlapped probe # se utiliza para contar el número de valores NA (ausentes) y no NA en la columna Probe_rs del objeto ann450k.</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">ann450k</span><span class="op">$</span><span class="va">Probe_rs</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 FALSE   TRUE 
 87018 398494 </code></pre>
</div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># probes without snp</span></span>
<span><span class="va">no.snp.probe</span> <span class="op">&lt;-</span> <span class="va">ann450k</span><span class="op">$</span><span class="va">Name</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">ann450k</span><span class="op">$</span><span class="va">Probe_rs</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#se seleccionan las sondas que tienen un identificador SNP asignado.</span></span>
<span><span class="va">snp.probe</span> <span class="op">&lt;-</span> <span class="va">ann450k</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">ann450k</span><span class="op">$</span><span class="va">Probe_rs</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="co">#snps with maf &lt;= 0.05</span></span>
<span><span class="co">#identifica las sondas en el objeto snp.probe que tienen </span></span>
<span><span class="co">#una frecuencia alélica mínima (MAF) menor o igual a 0.05 y almacena sus nombres en el vector snp5.probe. Estas sondas pueden ser de interés para estudios que se centren en variantes genéticas menos comunes.</span></span>
<span><span class="va">snp5.probe</span> <span class="op">&lt;-</span> <span class="va">snp.probe</span><span class="op">$</span><span class="va">Name</span><span class="op">[</span><span class="va">snp.probe</span><span class="op">$</span><span class="va">Probe_maf</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># filtre met</span></span>
<span><span class="co">## filtra el objeto met para mantener solo las filas que corresponden a las sondas sin identificador SNP y las sondas con MAF menor o igual a 0.05</span></span>
<span><span class="va">met</span> <span class="op">&lt;-</span> <span class="va">met</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">met</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">no.snp.probe</span>, <span class="va">snp5.probe</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="co">#remove no-further needed dataset</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">no.snp.probe</span>, <span class="va">probe</span>, <span class="va">probe.na</span>, <span class="va">snp.probe</span>, <span class="va">snp5.probe</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Removing probes that have been demonstrated to map to multiple places in the genome.</span></span>
<span><span class="co"># list adapted from https://www.tandfonline.com/doi/full/10.4161/epi.23470</span></span>
<span></span>
<span><span class="va">crs.reac</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="st">"cross_reactive_probe.chen2013.csv"</span><span class="op">)</span></span>
<span><span class="co">#extrae la columna "TargetID" del objeto crs.reac, excluyendo la primera fila.</span></span>
<span><span class="va">crs.reac</span> <span class="op">&lt;-</span> <span class="va">crs.reac</span><span class="op">$</span><span class="va">TargetID</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># filtre met (-which ) para excluye las sondas que coinciden con secuencias de ADN cruzadas o no específicas.</span></span>
<span><span class="va">met</span> <span class="op">&lt;-</span> <span class="va">met</span><span class="op">[</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">met</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html">%in%</a></span> <span class="va">crs.reac</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">bval1</span> <span class="op">&lt;-</span> <span class="va">met</span></span>
<span><span class="co">## converting beta values to m_values</span></span>
<span><span class="co">## m = log2(beta/1-beta)</span></span>
<span><span class="va">mval1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">met</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span><span class="op">(</span><span class="va">x</span><span class="op">/</span><span class="op">(</span><span class="fl">1</span><span class="op">-</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#_________Differential methylation analysis_____________#</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html">table</a></span><span class="op">(</span><span class="va">clinical1</span><span class="op">$</span><span class="va">paper_Histologic.grade</span>, <span class="va">clinical1</span><span class="op">$</span><span class="va">paper_mRNA.cluster</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>            
             Basal_squamous Luminal Luminal_infiltrated Luminal_papillary  ND
  High Grade            143      28                  79               120   4
  Low Grade               1       0                   0                20   0
  ND                      0       0                   1                 2   0
            
             Neuronal
  High Grade       20
  Low Grade         0
  ND                0</code></pre>
</div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># filtering and grouping</span></span>
<span><span class="co">#clinical &lt;- clinical[, c("barcode", "paper_Histologic.grade", "paper_mRNA.cluster")]</span></span>
<span><span class="va">clinical</span><span class="op">&lt;-</span><span class="va">clinical1</span></span>
<span><span class="va">clinical</span><span class="op">&lt;-</span> <span class="va">clinical</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/select.html">select</a></span><span class="op">(</span><span class="va">barcode</span>, <span class="va">paper_Histologic.grade</span>, <span class="va">paper_mRNA.cluster</span><span class="op">)</span></span>
<span><span class="co"># remove rows with missing values</span></span>
<span><span class="co">#clinical &lt;- na.omit(clinical)</span></span>
<span><span class="va">clinical</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tidyr/man/drop_na.html">drop_na</a></span><span class="op">(</span><span class="va">clinical</span><span class="op">)</span></span>
<span><span class="co">#clinical &lt;- clinical[-which(clinical$paper_mRNA.cluster == "ND"), ]</span></span>
<span><span class="va">clinical</span> <span class="op">&lt;-</span> <span class="va">clinical</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/filter.html">filter</a></span><span class="op">(</span><span class="va">paper_mRNA.cluster</span> <span class="op">!=</span> <span class="st">"ND"</span><span class="op">)</span></span>
<span><span class="co">#clinical &lt;- clinical[-which(clinical$paper_Histologic.grade == "ND"), ]</span></span>
<span><span class="va">clinical</span><span class="op">&lt;-</span><span class="va">clinical</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/filter.html">filter</a></span><span class="op">(</span><span class="va">paper_Histologic.grade</span> <span class="op">!=</span> <span class="st">"ND"</span><span class="op">)</span></span>
<span><span class="co">#clinical &lt;- clinical[which(clinical$paper_mRNA.cluster == "Luminal_papillary"), ]</span></span>
<span><span class="va">clinical</span><span class="op">&lt;-</span><span class="va">clinical</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/filter.html">filter</a></span><span class="op">(</span><span class="va">paper_mRNA.cluster</span> <span class="op">==</span> <span class="st">"Luminal_papillary"</span><span class="op">)</span></span>
<span><span class="va">barcode</span> <span class="op">&lt;-</span> <span class="va">clinical</span><span class="op">$</span><span class="va">barcode</span></span>
<span><span class="va">barcode</span><span class="op">&lt;-</span> <span class="va">clinical</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/select.html">select</a></span><span class="op">(</span><span class="va">barcode</span><span class="op">)</span></span>
<span><span class="co">########### # removing samples from meth matrixes</span></span>
<span><span class="va">bval</span><span class="op">&lt;-</span><span class="va">bval1</span></span>
<span><span class="va">bval</span> <span class="op">&lt;-</span> <span class="va">bval</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">bval</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">clinical</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">bval</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 296027    140</code></pre>
</div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mval</span><span class="op">&lt;-</span><span class="va">mval1</span></span>
<span><span class="va">mval</span> <span class="op">&lt;-</span> <span class="va">mval</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">mval</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">clinical</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">mval</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 296027    140</code></pre>
</div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Making sure about samples in clinical and matrixes and their order</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">mval</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">clinical</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
TRUE 
 140 </code></pre>
</div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">bval</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">clinical</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
TRUE 
 140 </code></pre>
</div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">clinical</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">bval</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">clinical</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">mval</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Making grouping variable</span></span>
<span><span class="co">#clinical$paper_Histologic.grade &lt;- as.factor(clinical$paper_Histologic.grade)</span></span>
<span><span class="va">clinical</span> <span class="op">&lt;-</span> <span class="va">clinical</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/mutate.html">mutate</a></span><span class="op">(</span>paper_Histologic.grade <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">paper_Histologic.grade</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># levels(clinical$paper_Histologic.grade) # re-orders levels of a factor</span></span>
<span><span class="co"># clinical$paper_Histologic.grade &lt;- relevel(clinical$paper_Histologic.grade, ref = "Low Grade")</span></span>
<span><span class="va">clinical</span> <span class="op">&lt;-</span> <span class="va">clinical</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/mutate.html">mutate</a></span><span class="op">(</span>paper_Histologic.grade <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html">relevel</a></span><span class="op">(</span><span class="va">paper_Histologic.grade</span>, ref <span class="op">=</span> <span class="st">"Low Grade"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#_____________ DMC analysis________________#</span></span>
<span><span class="co"># used to create a design matrix for a linear regression model</span></span>
<span><span class="co"># symbol indicates that you want to model the response variable (not specified here) </span></span>
<span><span class="co"># based on the predictor variable</span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">paper_Histologic.grade</span>, data <span class="op">=</span> <span class="va">clinical</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># fit the linear model  # MArrayLM: Microarray Linear Model Fit </span></span>
<span><span class="co"># Make sure the number of rows in mval matches the number of rows in design</span></span>
<span><span class="co"># Fit the linear model</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html">lmFit</a></span><span class="op">(</span><span class="va">mval</span>, <span class="va">design</span><span class="op">)</span></span>
<span><span class="co"># fit the linear model </span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html">lmFit</a></span><span class="op">(</span><span class="va">mval</span>, <span class="va">design</span><span class="op">)</span></span>
<span><span class="co"># para mejorar la precisión de las estimaciones y controlar el error de tipo I.</span></span>
<span><span class="va">fit2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span></span>
<span><span class="co"># extracting significantly methylated probes</span></span>
<span><span class="co"># aplicar los pasos de regularización empírica y ajuste de valores de p.</span></span>
<span><span class="va">deff.meth</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span><span class="op">(</span><span class="va">fit2</span>, coef<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html">ncol</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span>, sort.by<span class="op">=</span><span class="st">"p"</span>,number <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html">nrow</a></span><span class="op">(</span><span class="va">mval</span><span class="op">)</span>, adjust.method <span class="op">=</span> <span class="st">"BY"</span><span class="op">)</span></span>
<span><span class="co"># Visualization</span></span>
<span><span class="co"># plot the top 10 most significantly differentially methylated CpGs </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">deff.meth</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>, <span class="kw">function</span><span class="op">(</span><span class="va">cpg</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/minfi/man/plotCpg.html">plotCpg</a></span><span class="op">(</span><span class="va">bval</span>, cpg<span class="op">=</span><span class="va">cpg</span>, pheno<span class="op">=</span><span class="va">clinical</span><span class="op">$</span><span class="va">paper_Histologic.grade</span>, ylab <span class="op">=</span> <span class="st">"Beta values"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Met_ADN_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>$cg04117396
NULL

$cg16177647
NULL

$cg23797032
NULL

$cg10113582
NULL

$cg19024575
NULL

$cg02893550
NULL

$cg01449677
NULL

$cg23834688
NULL

$cg07866001
NULL

$cg17939040
NULL</code></pre>
</div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># making a volcano plot</span></span>
<span><span class="co">#making dataset se utiliza accede a la columna correspondiente al valor de p en el objeto fit2,</span></span>
<span><span class="co">#calcula el logaritmo negativo en base 1fit2 p.value of paper_Histologic.gradeHigh Grade</span></span>
<span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>foldchange <span class="op">=</span> <span class="va">fit</span><span class="op">[[</span><span class="st">"coefficients"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>, </span>
<span>                  logPvalue <span class="op">=</span>  <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">fit2</span><span class="op">[[</span><span class="st">"p.value"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># ver si el valor absoluto de la columna foldchange en dat es menor que 0.4. Esto devuelve un vector de valores TRUE o FALSE</span></span>
<span><span class="va">dat</span><span class="op">$</span><span class="va">threshold</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">dat</span><span class="op">$</span><span class="va">foldchange</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">0.4</span><span class="op">)</span></span>
<span><span class="co">#Visualization</span></span>
<span><span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TRUE"</span> <span class="op">=</span> <span class="st">"grey"</span>, <span class="st">"FALSE"</span> <span class="op">=</span> <span class="st">"blue"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">dat</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">foldchange</span>, y <span class="op">=</span> <span class="va">logPvalue</span>, color<span class="op">=</span><span class="va">threshold</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">.4</span>, size<span class="op">=</span><span class="fl">1.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">cols</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0.4</span>, colour<span class="op">=</span><span class="st">"#990000"</span>, linetype<span class="op">=</span><span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="op">-</span> <span class="fl">0.4</span>, colour<span class="op">=</span><span class="st">"#990000"</span>, linetype<span class="op">=</span><span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position<span class="op">=</span><span class="st">"none"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"Methylation difference (delta beta)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"-log10 p value"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Met_ADN_files/figure-html/unnamed-chunk-2-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># setting some annotation</span></span>
<span><span class="co"># para realizar una anotación de metilación de CpG en el objeto mval</span></span>
<span><span class="va">myAnnotation</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DMRcate/man/cpg.annotate.html">cpg.annotate</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">mval</span>, datatype <span class="op">=</span> <span class="st">"array"</span>, </span>
<span>                             what <span class="op">=</span> <span class="st">"M"</span>, </span>
<span>                             analysis.type <span class="op">=</span> <span class="st">"differential"</span>, </span>
<span>                             design <span class="op">=</span> <span class="va">design</span>, </span>
<span>                             contrasts <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                             coef <span class="op">=</span> <span class="st">"paper_Histologic.gradeHigh Grade"</span>, </span>
<span>                             arraytype <span class="op">=</span> <span class="st">"450K"</span>,</span>
<span>                             fdr <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span></span>
<span><span class="co">#el objeto myAnnotation contiene información anotada de metilación de CpG, incluyendo ubicaciones genómicas, estadísticas y metadatos adicionales relacionados con los CpG anotados.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">myAnnotation</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Formal class 'CpGannotated' [package "DMRcate"] with 1 slot
  ..@ ranges:Formal class 'GRanges' [package "GenomicRanges"] with 7 slots
  .. .. ..@ seqnames       :Formal class 'Rle' [package "S4Vectors"] with 4 slots
  .. .. .. .. ..@ values         : Factor w/ 22 levels "chr1","chr2",..: 1 2 3 4 5 6 7 8 9 10 ...
  .. .. .. .. ..@ lengths        : int [1:22] 30814 20969 15373 11252 13886 21203 17971 12475 6148 14891 ...
  .. .. .. .. ..@ elementMetadata: NULL
  .. .. .. .. ..@ metadata       : list()
  .. .. ..@ ranges         :Formal class 'IRanges' [package "IRanges"] with 6 slots
  .. .. .. .. ..@ start          : int [1:296027] 812539 834183 834295 834356 837536 838486 839752 840274 840618 841289 ...
  .. .. .. .. ..@ width          : int [1:296027] 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..@ NAMES          : chr [1:296027] "cg18147296" "cg13938959" "cg12445832" "cg23999112" ...
  .. .. .. .. ..@ elementType    : chr "ANY"
  .. .. .. .. ..@ elementMetadata: NULL
  .. .. .. .. ..@ metadata       : list()
  .. .. ..@ strand         :Formal class 'Rle' [package "S4Vectors"] with 4 slots
  .. .. .. .. ..@ values         : Factor w/ 3 levels "+","-","*": 3
  .. .. .. .. ..@ lengths        : int 296027
  .. .. .. .. ..@ elementMetadata: NULL
  .. .. .. .. ..@ metadata       : list()
  .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package "GenomeInfoDb"] with 4 slots
  .. .. .. .. ..@ seqnames   : chr [1:22] "chr1" "chr2" "chr3" "chr4" ...
  .. .. .. .. ..@ seqlengths : int [1:22] NA NA NA NA NA NA NA NA NA NA ...
  .. .. .. .. ..@ is_circular: logi [1:22] NA NA NA NA NA NA ...
  .. .. .. .. ..@ genome     : chr [1:22] NA NA NA NA ...
  .. .. ..@ elementMetadata:Formal class 'DFrame' [package "S4Vectors"] with 6 slots
  .. .. .. .. ..@ rownames       : NULL
  .. .. .. .. ..@ nrows          : int 296027
  .. .. .. .. ..@ elementType    : chr "ANY"
  .. .. .. .. ..@ elementMetadata: NULL
  .. .. .. .. ..@ metadata       : list()
  .. .. .. .. ..@ listData       :List of 4
  .. .. .. .. .. ..$ stat   : num [1:296027] -1.03 -1.91 -3.14 -3.62 3.29 ...
  .. .. .. .. .. ..$ diff   : num [1:296027] -0.035 -0.0835 -0.1329 -0.1832 0.1521 ...
  .. .. .. .. .. ..$ ind.fdr: num [1:296027] 0.50219 0.17593 0.02214 0.00824 0.01649 ...
  .. .. .. .. .. ..$ is.sig : logi [1:296027] FALSE FALSE FALSE FALSE FALSE FALSE ...
  .. .. ..@ elementType    : chr "ANY"
  .. .. ..@ metadata       : list()</code></pre>
</div>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># DMR analysis #  realiza un análisis de diferencial de metilación (DMR), utilizando la función dmrcate</span></span>
<span><span class="co"># o identification and extraction of differentially methylated regions</span></span>
<span><span class="va">DMRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DMRcate/man/dmrcate.html">dmrcate</a></span><span class="op">(</span><span class="va">myAnnotation</span>, lambda<span class="op">=</span><span class="fl">1000</span>, C<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="co"># sese utiliza para extraer las ubicaciones genómicas de los DMRs detectados en </span></span>
<span><span class="co"># forma de un objeto de la clase "GRanges" del paquete "GenomicRanges". Esto permite acceder y manipular fácilmente la información de los DMRs.</span></span>
<span><span class="va">results.ranges</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DMRcate/man/extractRanges.html">extractRanges</a></span><span class="op">(</span><span class="va">DMRs</span><span class="op">)</span></span>
<span><span class="co"># Estos metadatos proporcionan información detallada sobre los DMRs detectados, como el número de CpGs involucrados, </span></span>
<span><span class="co"># las estadísticas de diferenciación y los genes asociados. Puedes utilizar esta información para realizar un análisis más exhaustivo de los DMRs y comprender mejor su relevancia biológica.</span></span>
<span><span class="va">results.ranges</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 463 ranges and 8 metadata columns:
        seqnames              ranges strand |   no.cpgs min_smoothed_fdr
           &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;        &lt;numeric&gt;
    [1]     chr6   33244775-33247250      * |        56      8.42986e-88
    [2]    chr17   46651186-46653711      * |        12      2.96214e-57
    [3]    chr11     1325718-1328622      * |        12      7.12492e-42
    [4]    chr17   76794506-76795278      * |         5      4.00212e-59
    [5]     chr6   33048254-33049494      * |        11      8.58374e-59
    ...      ...                 ...    ... .       ...              ...
  [459]     chr7 157402637-157402673      * |         2      4.25760e-14
  [460]     chr7 128550623-128550812      * |         2      2.88595e-14
  [461]    chr16       833110-833250      * |         2      8.86884e-15
  [462]     chr6 100912072-100912168      * |         2      2.09966e-14
  [463]    chr11     2322808-2323059      * |         4      2.23375e-14
           Stouffer       HMFDR      Fisher    maxdiff   meandiff
          &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;  &lt;numeric&gt;  &lt;numeric&gt;
    [1] 7.89638e-34 2.55749e-03 1.70313e-37   0.278584  0.0947057
    [2] 3.33226e-40 2.06223e-05 3.47858e-36   0.434035  0.3332666
    [3] 5.48702e-33 5.19755e-06 1.71525e-30   0.427168  0.2552949
    [4] 7.25473e-29 8.96197e-10 3.54734e-30   0.581938  0.3804778
    [5] 1.33054e-26 8.37984e-07 1.56517e-28   0.406335  0.2358054
    ...         ...         ...         ...        ...        ...
  [459]  0.00173160   0.0111880  0.00288268 -0.1879305 -0.1575541
  [460]  0.00648757   0.0369501  0.01142196  0.1638965  0.1460685
  [461]  0.00767060   0.0357965  0.01289065  0.0729063  0.0527957
  [462]  0.03359317   0.0856047  0.05152012  0.1247401  0.1110737
  [463]  0.09790684   0.0918771  0.09052497 -0.0943266  0.0155559
             overlapping.genes
                   &lt;character&gt;
    [1]         B3GALT4, WDR46
    [2] HOXB-AS3, HOXB3, HOXB4
    [3]                 TOLLIP
    [4]                  USP36
    [5]     HLA-DPB1, HLA-DPA1
    ...                    ...
  [459]                 PTPRN2
  [460]                    KCP
  [461]                  MSLNL
  [462]                   SIM1
  [463]               C11orf21
  -------
  seqinfo: 22 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># visualization</span></span>
<span><span class="va">dmr.table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">results.ranges</span><span class="op">)</span></span>
<span><span class="co"># setting up variable for groupinh and color</span></span>
<span><span class="co"># Puedes utilizar estos códigos de color para asignar colores a diferentes elementos en tus visualizaciones</span></span>
<span><span class="va">pal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">8</span>,<span class="st">"Dark2"</span><span class="op">)</span></span>
<span><span class="va">groups</span> <span class="op">&lt;-</span> <span class="va">pal</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html">unique</a></span><span class="op">(</span><span class="va">clinical</span><span class="op">$</span><span class="va">paper_Histologic.grade</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">groups</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Factor-class.html">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">clinical</span><span class="op">$</span><span class="va">paper_Histologic.grade</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#setting up the genomic region </span></span>
<span><span class="va">gen</span> <span class="op">&lt;-</span> <span class="st">"hg19"</span></span>
<span><span class="co"># the index of the DMR that we will plot </span></span>
<span><span class="va">dmrIndex</span> <span class="op">&lt;-</span> <span class="fl">2</span></span>
<span><span class="co"># coordinates are stored under results.ranges[dmrIndex]</span></span>
<span><span class="va">chrom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">results.ranges</span><span class="op">[</span><span class="va">dmrIndex</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">start</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html">start</a></span><span class="op">(</span><span class="va">results.ranges</span><span class="op">[</span><span class="va">dmrIndex</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">end</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html">end</a></span><span class="op">(</span><span class="va">results.ranges</span><span class="op">[</span><span class="va">dmrIndex</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># add 25% extra space to plot</span></span>
<span><span class="va">minbase</span> <span class="op">&lt;-</span> <span class="va">start</span> <span class="op">-</span> <span class="op">(</span><span class="fl">0.25</span><span class="op">*</span><span class="op">(</span><span class="va">end</span><span class="op">-</span><span class="va">start</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">maxbase</span> <span class="op">&lt;-</span> <span class="va">end</span> <span class="op">+</span> <span class="op">(</span><span class="fl">0.25</span><span class="op">*</span><span class="op">(</span><span class="va">end</span><span class="op">-</span><span class="va">start</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># defining CpG islands track</span></span>
<span><span class="co"># download cpgislands for chromosome number 6 from ucsc</span></span>
<span><span class="va">chr6.cpg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="st">"chr6-cpg.csv"</span><span class="op">)</span></span>
<span><span class="co">#  que contiene información sobre las secuencias genómicas, las coordenadas y la orientación de los datos </span></span>
<span><span class="va">islandData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>seqnames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Rle-class.html">Rle</a></span><span class="op">(</span><span class="va">chr6.cpg</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span>, </span>
<span>                      ranges<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span>start<span class="op">=</span><span class="va">chr6.cpg</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>,</span>
<span>                                     end<span class="op">=</span><span class="va">chr6.cpg</span><span class="op">[</span>,<span class="fl">3</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                      strand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Rle-class.html">Rle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html">strand</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"*"</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html">nrow</a></span><span class="op">(</span><span class="va">chr6.cpg</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># DNAseI hypersensitive sites track</span></span>
<span><span class="co">#downloaded from ucsc</span></span>
<span><span class="va">chr6.dnase</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="st">"chr6-dnase.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dnaseData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>seqnames<span class="op">=</span><span class="va">chr6.dnase</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>,</span>
<span>                     ranges<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span>start<span class="op">=</span><span class="va">chr6.dnase</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>, end<span class="op">=</span><span class="va">chr6.dnase</span><span class="op">[</span>,<span class="fl">3</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                     strand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Rle-class.html">Rle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"*"</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html">nrow</a></span><span class="op">(</span><span class="va">chr6.dnase</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                     data<span class="op">=</span><span class="va">chr6.dnase</span><span class="op">[</span>,<span class="fl">5</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#Setting up the ideogram, genome and RefSeq tracks </span></span>
<span><span class="co"># representación ideográfica de un cromosoma en un genoma específico</span></span>
<span><span class="va">iTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Gviz/man/IdeogramTrack-class.html">IdeogramTrack</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="va">gen</span>, chromosome <span class="op">=</span> <span class="va">chrom</span>, name<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">chrom</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">gTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Gviz/man/GenomeAxisTrack-class.html">GenomeAxisTrack</a></span><span class="op">(</span>col<span class="op">=</span><span class="st">"black"</span>, cex<span class="op">=</span><span class="fl">1</span>, name<span class="op">=</span><span class="st">""</span>, fontcolor<span class="op">=</span><span class="st">"black"</span><span class="op">)</span></span>
<span><span class="co"># representa una región génica utilizando datos de NCBI RefSeq.</span></span>
<span><span class="va">rTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Gviz/man/UcscTrack.html">UcscTrack</a></span><span class="op">(</span>genome<span class="op">=</span><span class="va">gen</span>, chromosome<span class="op">=</span><span class="va">chrom</span>, track<span class="op">=</span><span class="st">"NCBI RefSeq"</span>, </span>
<span>                    from<span class="op">=</span><span class="va">minbase</span>, to<span class="op">=</span><span class="va">maxbase</span>, trackType<span class="op">=</span><span class="st">"GeneRegionTrack"</span>, </span>
<span>                    rstarts<span class="op">=</span><span class="st">"exonStarts"</span>, rends<span class="op">=</span><span class="st">"exonEnds"</span>, gene<span class="op">=</span><span class="st">"name"</span>, </span>
<span>                    symbol<span class="op">=</span><span class="st">"name2"</span>, transcript<span class="op">=</span><span class="st">"name"</span>, strand<span class="op">=</span><span class="st">"strand"</span>, </span>
<span>                    fill<span class="op">=</span><span class="st">"darkblue"</span>,stacking<span class="op">=</span><span class="st">"squish"</span>, name<span class="op">=</span><span class="st">"RefSeq"</span>, </span>
<span>                    showId<span class="op">=</span><span class="cn">TRUE</span>, geneSymbol<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># Ensure that the methylation data is ordered by chromosome and base position.</span></span>
<span><span class="co"># ordenar el dataframe ann450k por las columnas "chr" y "pos" en orden ascendente</span></span>
<span><span class="va">ann450kOrd</span> <span class="op">&lt;-</span> <span class="va">ann450k</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html">order</a></span><span class="op">(</span><span class="va">ann450k</span><span class="op">$</span><span class="va">chr</span>,<span class="va">ann450k</span><span class="op">$</span><span class="va">pos</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="co"># realiza una correspondencia entre los valores de la columna "Name" del dataframe ann450kOrd y los nombres de fila del dataframe bval</span></span>
<span><span class="va">bvalOrd</span> <span class="op">&lt;-</span> <span class="va">bval</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html">match</a></span><span class="op">(</span><span class="va">ann450kOrd</span><span class="op">$</span><span class="va">Name</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">bval</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="co">################### Create the data tracks ###################</span></span>
<span><span class="co"># create genomic ranges object from methylation data</span></span>
<span><span class="co"># cpgData contiene información sobre los sitios CpG (CpG dinucleótidos) del genoma.</span></span>
<span><span class="va">cpgData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>seqnames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Rle-class.html">Rle</a></span><span class="op">(</span><span class="va">ann450kOrd</span><span class="op">$</span><span class="va">chr</span><span class="op">)</span>,</span>
<span>                   ranges<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span>start<span class="op">=</span><span class="va">ann450kOrd</span><span class="op">$</span><span class="va">pos</span>, end<span class="op">=</span><span class="va">ann450kOrd</span><span class="op">$</span><span class="va">pos</span><span class="op">)</span>, <span class="co">#define los intervalos de los sitios CpG</span></span>
<span>                   strand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Rle-class.html">Rle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"*"</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html">nrow</a></span><span class="op">(</span><span class="va">ann450kOrd</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,<span class="co">#indica la dirección del sitio CpG. En este caso, se utiliza el valor "*" para indicar que la dirección es desconocida.</span></span>
<span>                   betas<span class="op">=</span><span class="va">bvalOrd</span><span class="op">)</span> <span class="co"># los valores de beta asociados a cada sitio CpG</span></span>
<span><span class="co"># methylation data track # representa una pista de datos de metilación de ADN.</span></span>
<span><span class="co"># una pista de datos que muestra la metilación de ADN en diferentes grupos a lo </span></span>
<span><span class="co">#largo de un cromosoma. Los datos se representan utilizando valores de beta y se visualizan como líneas y puntos en la pista.</span></span>
<span><span class="va">methTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Gviz/man/DataTrack-class.html">DataTrack</a></span><span class="op">(</span>range<span class="op">=</span><span class="va">cpgData</span>, </span>
<span>                       groups<span class="op">=</span><span class="va">clinical</span><span class="op">$</span><span class="va">paper_Histologic.grade</span>, <span class="co"># change this if your groups are diffrent</span></span>
<span>                       genome <span class="op">=</span> <span class="va">gen</span>,</span>
<span>                       chromosome<span class="op">=</span><span class="va">chrom</span>,</span>
<span>                       ylim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.05</span>,<span class="fl">1.05</span><span class="op">)</span>,</span>
<span>                       col<span class="op">=</span><span class="va">pal</span>,</span>
<span>                       type<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"a"</span>,<span class="st">"p"</span><span class="op">)</span>, </span>
<span>                       name<span class="op">=</span><span class="st">"DNA Meth.\n(beta value)"</span>,</span>
<span>                       background.panel<span class="op">=</span><span class="st">"white"</span>, </span>
<span>                       legend<span class="op">=</span><span class="cn">TRUE</span>, </span>
<span>                       cex.title<span class="op">=</span><span class="fl">0.8</span>,</span>
<span>                       cex.axis<span class="op">=</span><span class="fl">0.8</span>, </span>
<span>                       cex.legend<span class="op">=</span><span class="fl">0.8</span><span class="op">)</span></span>
<span><span class="co"># CpG island track  pista de anotaciones que muestra la ubicación y extensión de las islas CpG en el genoma. </span></span>
<span><span class="va">islandTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Gviz/man/AnnotationTrack-class.html">AnnotationTrack</a></span><span class="op">(</span>range<span class="op">=</span><span class="va">islandData</span>, genome<span class="op">=</span><span class="va">gen</span>, name<span class="op">=</span><span class="st">"CpG Is."</span>, </span>
<span>                               chromosome<span class="op">=</span><span class="va">chrom</span>,fill<span class="op">=</span><span class="st">"darkgreen"</span><span class="op">)</span></span>
<span><span class="co"># DNaseI hypersensitive site data track</span></span>
<span><span class="co"># una pista de datos que muestra la ubicación y la intensidad de los sitios de hipersensibilidad a la DNasa I en el geno</span></span>
<span><span class="va">dnaseTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Gviz/man/DataTrack-class.html">DataTrack</a></span><span class="op">(</span>range<span class="op">=</span><span class="va">dnaseData</span>, genome<span class="op">=</span><span class="va">gen</span>, name<span class="op">=</span><span class="st">"DNAseI"</span>, </span>
<span>                        type<span class="op">=</span><span class="st">"gradient"</span>, chromosome<span class="op">=</span><span class="va">chrom</span><span class="op">)</span></span>
<span><span class="co"># DMR position data track</span></span>
<span><span class="va">dmrTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Gviz/man/AnnotationTrack-class.html">AnnotationTrack</a></span><span class="op">(</span>start<span class="op">=</span><span class="va">start</span>, end<span class="op">=</span><span class="va">end</span>, genome<span class="op">=</span><span class="va">gen</span>, name<span class="op">=</span><span class="st">"DMR"</span>, </span>
<span>                            chromosome<span class="op">=</span><span class="va">chrom</span>,fill<span class="op">=</span><span class="st">"darkred"</span><span class="op">)</span></span>
<span><span class="co"># Set up the track list and indicate the relative sizes of the different tracks. </span></span>
<span><span class="co"># Finally, draw the plot using the plotTracks function</span></span>
<span><span class="va">tracks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">iTrack</span>, <span class="va">gTrack</span>, <span class="va">methTrack</span>, <span class="va">dmrTrack</span>, <span class="va">islandTrack</span>, <span class="va">dnaseTrack</span>,</span>
<span>               <span class="va">rTrack</span><span class="op">)</span></span>
<span><span class="va">sizes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">5</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span> <span class="co"># set up the relative sizes of the tracks</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grDevices/png.html">tiff</a></span><span class="op">(</span> filename <span class="op">=</span> <span class="st">"dmr.tiff"</span>, width <span class="op">=</span> <span class="fl">15</span>, height <span class="op">=</span> <span class="fl">10</span>, units <span class="op">=</span> <span class="st">"in"</span>, res <span class="op">=</span> <span class="fl">400</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>



</main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./index.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Bienvenido</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
  </div>
</nav>
</div> <!-- /content -->



</body></html>